File: master_wallet/ios/TigerMasterWallet/Sources/Services/ThemeService.swift
// MasterWallet Theme Service (iOS)
// Light/Dark theme switching
// Production-ready

import UIKit

class ThemeService {
    
    static let shared = ThemeService()
    
    private let themeKey = "theme_mode"
    private var currentTheme: String = "light"
    private var listeners: [ThemeChangeListener] = []
    
    // MARK: - Colors
    
    struct ThemeColors {
        let background: UIColor
        let surface: UIColor
        let surfaceElevated: UIColor
        let primary: UIColor
        let primaryVariant: UIColor
        let secondary: UIColor
        let accent: UIColor
        let text: UIColor
        let textSecondary: UIColor
        let textMuted: UIColor
        let heading: UIColor
        let link: UIColor
        let border: UIColor
        let success: UIColor
        let warning: UIColor
        let error: UIColor
        let onPrimary: UIColor
        let isDark: Bool
    }
    
    // MARK: - Initialize
    
    private init() {}
    
    func initialize() {
        currentTheme = UserDefaults.standard.string(forKey: themeKey) ?? getSystemTheme()
        applyTheme()
    }
    
    // MARK: - Theme Control
    
    func getTheme() -> String {
        return currentTheme
    }
    
    func isDarkMode() -> Bool {
        return currentTheme == "dark"
    }
    
    func setTheme(_ theme: String) {
        guard theme == "light" || theme == "dark" else { return }
        
        currentTheme = theme
        UserDefaults.standard.set(theme, forKey: themeKey)
        
        applyTheme()
        notifyListeners()
    }
    
    func toggleTheme() {
        setTheme(currentTheme == "light" ? "dark" : "light")
    }
    
    // MARK: - Apply Theme
    
    private func applyTheme() {
        let style: UIUserInterfaceStyle = currentTheme == "dark" ? .dark : .light
        
        if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene {
            windowScene.windows.forEach { window in
                window.overrideUserInterfaceStyle = style
            }
        }
        
        // Apply global tint
        UIView.appearance().tintColor = getThemeColors().primary
    }
    
    // MARK: - Theme Colors
    
    func getThemeColors() -> ThemeColors {
        if currentTheme == "dark" {
            return ThemeColors(
                background: UIColor(hex: 0x0A0A0A),
                surface: UIColor(hex: 0x1A1A1A),
                surfaceElevated: UIColor(hex: 0x242424),
                primary: UIColor(hex: 0x3B82F6),
                primaryVariant: UIColor(hex: 0x2563EB),
                secondary: UIColor(hex: 0x6366F1),
                accent: UIColor(hex: 0x8B5CF6),
                text: UIColor(hex: 0xE5E5E5),
                textSecondary: UIColor(hex: 0xA3A3A3),
                textMuted: UIColor(hex: 0x737373),
                heading: UIColor(hex: 0xF5F5F5),
                link: UIColor(hex: 0x60A5FA),
                border: UIColor(hex: 0x333333),
                success: UIColor(hex: 0x22C55E),
                warning: UIColor(hex: 0xF59E0B),
                error: UIColor(hex: 0xEF4444),
                onPrimary: UIColor(hex: 0xFFFFFFFF),
                isDark: true
            )
        } else {
            return ThemeColors(
                background: UIColor(hex: 0xFFFFFF),
                surface: UIColor(hex: 0xF9FAFB),
                surfaceElevated: UIColor(hex: 0xFFFFFF),
                primary: UIColor(hex: 0x3B82F6),
                primaryVariant: UIColor(hex: 0x2563EB),
                secondary: UIColor(hex: 0x6366F1),
                accent: UIColor(hex: 0x8B5CF6),
                text: UIColor(hex: 0x171717),
                textSecondary: UIColor(hex: 0x525252),
                textMuted: UIColor(hex: 0xA3A3A3),
                heading: UIColor(hex: 0x0A0A0A),
                link: UIColor(hex: 0x2563EB),
                border: UIColor(hex: 0xE5E5E5),
                success: UIColor(hex: 0x16A34A),
                warning: UIColor(hex: 0xD97706),
                error: UIColor(hex: 0xDC2626),
                onPrimary: UIColor(hex: 0xFFFFFFFF),
                isDark: false
            )
        }
    }
    
    func getColor(named name: String) -> UIColor {
        let colors = getThemeColors()
        
        switch name {
        case "background": return colors.background
        case "surface": return colors.surface
        case "surfaceElevated": return colors.surfaceElevated
        case "primary": return colors.primary
        case "primaryVariant": return colors.primaryVariant
        case "secondary": return colors.secondary
        case "accent": return colors.accent
        case "text": return colors.text
        case "textSecondary": return colors.textSecondary
        case "textMuted": return colors.textMuted
        case "heading": return colors.heading
        case "link": return colors.link
        case "border": return colors.border
        case "success": return colors.success
        case "warning": return colors.warning
        case "error": return colors.error
        case "onPrimary": return colors.onPrimary
        default: return colors.text
        }
    }
    
    // MARK: - Listeners
    
    func addListener(_ listener: ThemeChangeListener) {
        if !listeners.contains(where: { $0 === listener }) {
            listeners.append(listener)
        }
    }
    
    func removeListener(_ listener: ThemeChangeListener) {
        listeners.removeAll { $0 === listener }
    }
    
    private func notifyListeners() {
        for listener in listeners {
            listener.onThemeChanged(currentTheme)
        }
    }
    
    // MARK: - Private Methods
    
    private func getSystemTheme() -> String {
        let style = UITraitCollection.current.userInterfaceStyle
        return style == .dark ? "dark" : "light"
    }
}

// MARK: - Protocol

protocol ThemeChangeListener: AnyObject {
    func onThemeChanged(_ theme: String)
}

// MARK: - UIColor Extension

extension UIColor {
    convenience init(hex: Int, alpha: CGFloat = 1.0) {
        let red = CGFloat((hex >> 16) & 0xFF) / 255.0
        let green = CGFloat((hex >> 8) & 0xFF) / 255.0
        let blue = CGFloat(hex & 0xFF) / 255.0
        self.init(red: red, green: green, blue: blue, alpha: alpha)
    }
}

File: master_wallet/android/app/src/main/java/com/tigermaster/services/ThemeService.kt
package com.tigermaster.services

import android.content.Context
import android.content.res.Configuration
import androidx.appcompat.app.AppCompatDelegate
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey

/**
 * MasterWallet Theme Service (Android)
 * Light/Dark theme switching for all pages
 * Production-ready
 */
class ThemeService(private val context: Context) {
    
    companion object {
        private const val PREFS_NAME = "theme_prefs"
        private const val KEY_THEME = "theme_mode"
    }
    
    private val masterKey: MasterKey by lazy {
        MasterKey.Builder(context)
            .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
            .build()
    }
    
    private val encryptedPrefs by lazy {
        EncryptedSharedPreferences.create(
            context,
            PREFS_NAME,
            masterKey,
            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
        )
    }
    
    private var currentTheme: String = "light"
    private var listeners: MutableList<ThemeChangeListener> = mutableListOf()
    
    /**
     * Initialize the theme service
     */
    fun initialize() {
        // Load saved theme
        currentTheme = encryptedPrefs.getString(KEY_THEME, getSystemTheme()) ?: "light"
        
        // Apply theme
        applyTheme()
    }
    
    /**
     * Get current theme
     */
    fun getTheme(): String = currentTheme
    
    /**
     * Check if dark mode
     */
    fun isDarkMode(): Boolean = currentTheme == "dark"
    
    /**
     * Set theme
     */
    fun setTheme(theme: String) {
        if (theme != "light" && theme != "dark") return
        
        currentTheme = theme
        encryptedPrefs.edit().putString(KEY_THEME, theme).apply()
        
        applyTheme()
        
        // Notify listeners
        notifyListeners()
    }
    
    /**
     * Toggle theme
     */
    fun toggleTheme() {
        setTheme(if (currentTheme == "light") "dark" else "light")
    }
    
    /**
     * Apply theme to the app
     */
    private fun applyTheme() {
        val nightMode = if (currentTheme == "dark") {
            AppCompatDelegate.MODE_NIGHT_YES
        } else {
            AppCompatDelegate.MODE_NIGHT_NO
        }
        
        AppCompatDelegate.setDefaultNightMode(nightMode)
    }
    
    /**
     * Get system theme preference
     */
    private fun getSystemTheme(): String {
        val nightModeFlags = context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        return when (nightModeFlags) {
            Configuration.UI_MODE_NIGHT_YES -> "dark"
            Configuration.UI_MODE_NIGHT_NO -> "light"
            else -> "light"
        }
    }
    
    /**
     * Add theme change listener
     */
    fun addListener(listener: ThemeChangeListener) {
        if (!listeners.contains(listener)) {
            listeners.add(listener)
        }
    }
    
    /**
     * Remove theme change listener
     */
    fun removeListener(listener: ThemeChangeListener) {
        listeners.remove(listener)
    }
    
    /**
     * Notify all listeners
     */
    private fun notifyListeners() {
        for (listener in listeners) {
            listener.onThemeChanged(currentTheme)
        }
    }
    
    /**
     * Get theme colors
     */
    fun getThemeColors(): ThemeColors {
        return if (currentTheme == "dark") {
            ThemeColors(
                background = 0xFF0A0A0A.toInt(),
                surface = 0xFF1A1A1A.toInt(),
                surfaceElevated = 0xFF242424.toInt(),
                primary = 0xFF3B82F6.toInt(),
                primaryVariant = 0xFF2563EB.toInt(),
                secondary = 0xFF6366F1.toInt(),
                accent = 0xFF8B5CF6.toInt(),
                text = 0xFFE5E5E5.toInt(),
                textSecondary = 0xFFA3A3A3.toInt(),
                textMuted = 0xFF737373.toInt(),
                heading = 0xFFF5F5F5.toInt(),
                link = 0xFF60A5FA.toInt(),
                border = 0xFF333333.toInt(),
                success = 0xFF22C55E.toInt(),
                warning = 0xFFF59E0B.toInt(),
                error = 0xFFEF4444.toInt(),
                onPrimary = 0xFFFFFFFF.toInt(),
                isDark = true
            )
        } else {
            ThemeColors(
                background = 0xFFFFFFFF.toInt(),
                surface = 0xFFF9FAFB.toInt(),
                surfaceElevated = 0xFFFFFFFF.toInt(),
                primary = 0xFF3B82F6.toInt(),
                primaryVariant = 0xFF2563EB.toInt(),
                secondary = 0xFF6366F1.toInt(),
                accent = 0xFF8B5CF6.toInt(),
                text = 0xFF171717.toInt(),
                textSecondary = 0xFF525252.toInt(),
                textMuted = 0xFFA3A3A3.toInt(),
                heading = 0xFF0A0A0A.toInt(),
                link = 0xFF2563EB.toInt(),
                border = 0xFFE5E5E5.toInt(),
                success = 0xFF16A34A.toInt(),
                warning = 0xFFD97706.toInt(),
                error = 0xFFDC2626.toInt(),
                onPrimary = 0xFFFFFFFF.toInt(),
                isDark = false
            )
        }
    }
    
    /**
     * Get color for resource name
     */
    fun getColor(colorName: String): Int {
        return when (colorName) {
            "background" -> getThemeColors().background
            "surface" -> getThemeColors().surface
            "surface_elevated" -> getThemeColors().surfaceElevated
            "primary" -> getThemeColors().primary
            "primary_variant" -> getThemeColors().primaryVariant
            "secondary" -> getThemeColors().secondary
            "accent" -> getThemeColors().accent
            "text" -> getThemeColors().text
            "text_secondary" -> getThemeColors().textSecondary
            "text_muted" -> getThemeColors().textMuted
            "heading" -> getThemeColors().heading
            "link" -> getThemeColors().link
            "border" -> getThemeColors().border
            "success" -> getThemeColors().success
            "warning" -> getThemeColors().warning
            "error" -> getThemeColors().error
            "on_primary" -> getThemeColors().onPrimary
            else -> getThemeColors().text
        }
    }
}

/**
 * Theme change listener interface
 */
interface ThemeChangeListener {
    fun onThemeChanged(theme: String)
}

/**
 * Theme colors data class
 */
data class ThemeColors(
    val background: Int,
    val surface: Int,
    val surfaceElevated: Int,
    val primary: Int,
    val primaryVariant: Int,
    val secondary: Int,
    val accent: Int,
    val text: Int,
    val textSecondary: Int,
    val textMuted: Int,
    val heading: Int,
    val link: Int,
    val border: Int,
    val success: Int,
    val warning: Int,
    val error: Int,
    val onPrimary: Int,
    val isDark: Boolean
)
